Readme.md

```diff
@@ -553,3 +617,212 @@ Example:
+
+```
+rol    reg1,reg2,Imm
+ror    reg1,reg2,Imm
+```
+
+Rotates reg2 left/right by Imm and stores the result in reg1. A combination of `sll`, `srl` and `or` is used.
+
+Additionally, there are upper and lower versions for many two opcode macros. They have the same names and parameters as the normal versions, but `.u` or `.l` is appended at the end of the name.
+For example, `li.u` will output the upper half of the `li` macro, and `li.l` will output the lower half. The following macros support this: `li`,`la`,`lb`,`lbu`,`lhu`,`lw`,`sb`,`sh`,`sw`
+
+This can be used when the two halves of the macros need to be used in nonconsecutive positions, for example:
+
+```
+li.u  a0,address
+jal   function
+li.l  a0,address
+```
+
+## 5.2 Assembler-defined ARM macros
+
+The assembler will automatically convert the arguments between the following opcodes if possible:
+
+```
+mov <-> mvn
+bic <-> and
+cmp <-> cmn
+```
+
+E.g., `mov r0,-1` will be assembled as `mvn r0,0`
+
+Additionally, `ldr rx,=immediate` can be used to load a 32-bit immediate. The assembler will try to convert it into a mov/mvn instruction if possible. Otherwise, it will be stored in the nearest pool (see the .pool directive). `add rx,=immediate` can be used as a PC-relative add and will be assembled as `add rx,r15,(immediate-.-8)`
+
+
+## 5.3 User defined macros
+
+The assembler allows the creation of custom macros. This is an example macro, a recreation of the builtin MIPS macro `li`:
+
+```
+.macro myli,dest,value
+   .if value < 0x10000
+      ori   dest,r0,value
+   .elseif (value & 0xFFFF8000) == 0xFFFF8000
+      addiu dest,r0,value & 0xFFFF
+   .elseif (value & 0xFFFF) == 0
+      lui   dest,value >> 16
+   .else
+      lui   dest,value >> 16 + (value & 0x8000 != 0)
+      addiu dest,dest,value & 0xFFFF
+   .endif
+.endmacro
+```
+
+The macro has to be initiated by a `.macro` directive. The first argument is the macro name, followed by a variable amount of arguments. The code inside the macro can be anything, and it can even call other macros (up to a nesting level of 128 calls). The macro is terminated by a `.endmacro` directive. It is not assembled when it is defined, but other code can call it from then on. All arguments are simple text replacements, so they can be anything from a number to a whole instruction parameter list. The macro is then invoked like this:
+
+```
+myli  a0,0xFFEEDDCC
+```
+
+In this case, the code will assemble to the following:
+
+```
+lui   a0,0xFFEF
+addiu a0,a0,0xDDCC
+```
+
+Like all the other code, any equs are inserted before they are resolved.
+
+Macros can also contain local labels that are changed to an unique name. Global labels, however, are unaffected by this. The label name is prefixed by the macro name and a counter id. This label:
+
+```
+.macro test
+  @@MainLoop:
+.endmacro
+```
+
+will therefore be changed to:
+
+```
+@@test_00000000_mainloop
+```
+
+Each call of the macro will increase the counter.
+
+# 6. Built-in functions
```

Comment:
I think it would be better to put this under the "Math parser" section as they are technically part of the expression feature. In addition, I think it would go to rename that section to "Expression parser".